src/dlgButtonSelect.cpp

```diff
@@ -0,0 +1,233 @@
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+
+
+#include "dlgButtonSelect.h"
+
+
+#include "Host.h"
+#include "HostManager.h"
+#include "mudlet.h"
+
+#include "pre_guard.h"
+#include <QFileDialog>
+#include <QInputDialog>
+#include <QMessageBox>
+#include <QPainter>
+#include <QtUiTools>
+#include <QGamepad>
+#include "post_guard.h"
+
+
+#define _DEBUG_
+
+dlgButtonSelect::dlgButtonSelect(QWidget * parent)
+: QDialog( parent )
+{
+    setupUi( this );
+
+    connect(buttonBox, SIGNAL(accepted()), this, SLOT(accept()));
+    connect(buttonBox, SIGNAL(rejected()), this, SLOT(reject()));
+
+    connect(QGamepadManager::instance(), &QGamepadManager::gamepadButtonPressEvent, this, slot_gamepadButtonPress);
+    connect(QGamepadManager::instance(), &QGamepadManager::gamepadButtonReleaseEvent, this, slot_gamepadButtonRelease);
+    connect(QGamepadManager::instance(), &QGamepadManager::gamepadAxisEvent, this, slot_gamepadAxisEvent);
+
+    connect(pressRadio, SIGNAL(pressed()), this, SLOT(slot_pressRadioPressed()));
+    connect(releaseRadio, SIGNAL(pressed()), this, SLOT(slot_releaseRadioPressed()));
+    connect(repickButton, SIGNAL(clicked()), this, SLOT(slot_repickButtonClicked()));
+
+    buttonSelected = false;
+}
+
+void dlgButtonSelect::slot_pressRadioPressed()
+{
+    pressThresholdSlider->setEnabled(true);
+}
+
+void dlgButtonSelect::slot_releaseRadioPressed()
+{
+    pressThresholdSlider->setDisabled(true);
+}
+
+void dlgButtonSelect::slot_repickButtonClicked()
+{
+    buttonSelected = false;
+    tabWidget->setCurrentIndex(0);
+    deviceEdit->setText("0");
+    buttonEdit->setText("0");
+    pressRadio->click();
+    pressThresholdSlider->setValue(75);
+    repickButton->setDisabled(true);
+}
+
+void dlgButtonSelect::slot_gamepadButtonPress(int deviceId, QGamepadManager::GamepadButton button, double value)
+{
+    lastEventLabel->setText(QString("Device %1: Button %2 was pressed to the %3 value").arg(deviceId).arg(button).arg(value));
+
+    if ( !buttonSelected ){
+        deviceEdit->setText(QString("%1").arg(deviceId));
+        tabWidget->setCurrentIndex(0);
+        buttonEdit->setText(QString("%1").arg(button));
+        buttonSelected = true;
+        repickButton->setEnabled(true);
+        return;
+    }
+
+    if ( button == 12 ){// dpad U
```

Comment:
Yeah. I saw that they have enums in place, just didn't feel the need to bother at that particular moment since I know which button is which from experience. I'll swap them over for enums before we end up merging this.